src/game/common/sparsematchfinder.h

```diff
@@ -22,7 +22,7 @@ template<typename Type, typename Key> class SparseMatchFinder
     class MapHelper
     {
     public:
-        bool operator()(const Key &left, const Key &right) const { return left == right; }
+        bool operator()(const Key &left, const Key &right) const { return left != right; }
```

Comment:
This is not correct, we should implement this the way the original did it (see function 006F1520 in ZH worldbuilder) to avoid any risk of bugs.